src/basic/time-util.c

```diff
@@ -429,18 +429,36 @@ char *format_timestamp_relative(char *buf, size_t l, usec_t t) {
                 s = "left";
         }
 
-        if (d >= USEC_PER_YEAR)
-                snprintf(buf, l, USEC_FMT " years " USEC_FMT " months %s",
-                         d / USEC_PER_YEAR,
-                         (d % USEC_PER_YEAR) / USEC_PER_MONTH, s);
-        else if (d >= USEC_PER_MONTH)
-                snprintf(buf, l, USEC_FMT " months " USEC_FMT " days %s",
-                         d / USEC_PER_MONTH,
-                         (d % USEC_PER_MONTH) / USEC_PER_DAY, s);
-        else if (d >= USEC_PER_WEEK)
-                snprintf(buf, l, USEC_FMT " weeks " USEC_FMT " days %s",
-                         d / USEC_PER_WEEK,
-                         (d % USEC_PER_WEEK) / USEC_PER_DAY, s);
+        if (d >= USEC_PER_YEAR) {
+                usec_t years = d / USEC_PER_YEAR;
+                usec_t months = (d % USEC_PER_YEAR) / USEC_PER_MONTH;
+                snprintf(buf, l, USEC_FMT " %s " USEC_FMT " %s %s",
+                         years,
+                         years == 1 ? "year" : "years",
+                         months,
+                         months == 1 ? "month" : "months",
+                         s);
+        }
+        else if (d >= USEC_PER_MONTH) {
```

Comment:
coding style: we generally put `}` and `else` on the same line

Reply:
Thanks for the input. Now that this is merged, what's the preferred way of fixing this? Shall I create a new PR for that purpose?

Reply:
I see @yuwata is fixing this in https://github.com/systemd/systemd/pull/19984.